Fixed an issue with redefining global Proc's ! method

smileart opened the pull request:
fix: an issue with proc's ! operator (https://is.gd/N0K9yn)
chore: bump the major version 0.1.0 → 1.0.0

Caught an issue using [json_schemer](https://github.com/davishmcclurg/json_schemer) which uses [ecma-re-validator](https://github.com/gjtorikian/ecma-re-validator), which depends on [pry-0.12.2](https://github.com/pry/pry) which uses ! on a block and unintentionally calls TestRocket's redefined version and crushes trying to use the result (`nil`).

Had to define TestRocket functionality through Ruby refinements to isolate the scope, therefore the usage changed and now it's necessary to use `using TestRocket` explicit syntax where using the tests. Hence the backwards compatibility is impossible and major version number bump was necessary (which was long expected anyway https://github.com/peterc/testrocket/issues/8).

Thanks for the project, really love it and would appreciate a fix and releasing this and other versions to https://rubygems.org and tagging the versions on GitHub. Thank you! 🙏

peterc commented on testrocket.gemspec:
```diff
@@ -1,21 +1,21 @@
-  s.authors     = ["Peter Cooper","Christoph Grabo"]
-  s.email       = ["git@peterc.org","chris@dinarrr.com"]
-  s.homepage    = "http://github.com/peterc/testrocket"
+  s.authors     = %w[Peter Cooper Christoph Grabo]
```
This needs to go back to the original, as it creates an array of four elements due to the spaces.

smileart replied:
Oh my! Sure thing. Gonna change it back.

peterc commented:
I've left a comment on a minor bug, but otherwise I'm not *against* this idea. I do need to let it sit with me for a bit though as well as give it a proper test of my own. But all being well, I think I'll come back to this in about a week and most likely accept this approach :-) Thanks!

smileart commented:
Thanks a lot for your time and consideration. I fixed my f*up in the second commit and changed a few other minor things for consistency purposes.

BTW, here's a minimal code sample to reproduce the behaviour:

````ruby
require 'testrocket'

ok = ->() { nil }
p !ok
````

If we comment or delete the requirement of `testrocket`, the code works as expected. If we leave it, it calls `testrocket`, which is not desired behaviour.ss

peterc commented:
OK, so I've approved and merged this locally but GitHub is being weird and not letting me push it up. Getting a very odd error "denied to omadeploy".. despite "omadeploy" not meaning anything, zero Google results, and not a word located in the entire repo either. Going to assume GitHub is brain farting and will try again later.

smileart commented:
Any updates on this one? Perhaps it (GitHub) should be working by now. Just a friendly reminder, no pressure. 👐

peterc commented:
Nope, I just tried again and got "remote: Permission to peterc/testrocket.git denied to omadeploy."

I can push to all of my other repos fine and have no idea what "omadeploy" even means. It literally has zero Google results(!)

What I might try next is using GitHub's UI to merge the pull request then make my documentation changes separately after.

peterc commented:
OK, take a look now and see if you're happy with everything. Using GitHub's UI instead seemed to work.

smileart commented:
Seems alright. Now I can switch to official RubyGems version (1.0.0) without having those Pry exceptions. Thanks a lot! 🙏
